fields.go

```diff
@@ -0,0 +1,156 @@
+// Go MySQL Driver - A MySQL-Driver for Go's database/sql package
+//
+// Copyright 2017 The Go-MySQL-Driver Authors. All rights reserved.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package mysql
+
+import (
+	"database/sql"
+	"reflect"
+	"time"
+)
+
+var typeDatabaseName = map[fieldType]string{
+	fieldTypeDecimal:    "DECIMAL",
+	fieldTypeTiny:       "TINYINT",
+	fieldTypeShort:      "SMALLINT",
+	fieldTypeLong:       "INT",
+	fieldTypeFloat:      "FLOAT",
+	fieldTypeDouble:     "DOUBLE",
+	fieldTypeNULL:       "NULL",
+	fieldTypeTimestamp:  "TIMESTAMP",
+	fieldTypeLongLong:   "BIGINT",
+	fieldTypeInt24:      "MEDIUMINT",
+	fieldTypeDate:       "DATE",
+	fieldTypeTime:       "TIME",
+	fieldTypeDateTime:   "DATETIME",
+	fieldTypeYear:       "YEAR",
+	fieldTypeNewDate:    "DATE",
+	fieldTypeVarChar:    "VARCHAR",
+	fieldTypeBit:        "BIT",
+	fieldTypeJSON:       "JSON",
+	fieldTypeNewDecimal: "DECIMAL",
+	fieldTypeEnum:       "ENUM",
+	fieldTypeSet:        "SET",
+	fieldTypeTinyBLOB:   "TINYBLOB",
+	fieldTypeMediumBLOB: "MEDIUMBLOB",
+	fieldTypeLongBLOB:   "LONGBLOB",
+	fieldTypeBLOB:       "BLOB",
+	fieldTypeVarString:  "VARSTRING", // correct?
```

Comment:
I didn't see these in MySQL anywhere. The docs, e.g. https://dev.mysql.com/doc/refman/5.7/en/string-types.html doesn't mention them either. I don't think they are correct, should probably be varchar.

Reply:
https://github.com/PyMySQL/PyMySQL/blob/e027812ab0eab4d0557fa2ab66520d5eb38fc28f/pymysql/connections.py#L1502-L1507

There are no document about it.
After seeing some response, I decided to handle charsetnr == 63 is bytes, not string.

Reply:
Of course, string can contain bytes.